### DISCUSSION TITLE
Routing key doesn't work

### DISCUSSION THREAD

**Discussion by dormanze** [rabbitmq/rabbitmq-server#6441] [Questions]
Hello everyone.
My Queue Information :
![image](https://user-images.githubusercontent.com/37854724/203283056-c3dd63fe-38fb-4397-849a-115fe4838567.png)
It has binding relationship . but when I use the routing key to push messages, it does not work.
![image](https://user-images.githubusercontent.com/37854724/203284353-75780047-6a31-409c-a458-b3cdcbf64423.png)
Later, I created two queues manually and bound this routing key and push message,it is work for new queues.
![image](https://user-images.githubusercontent.com/37854724/203289660-f2da1cbb-595f-4e10-beab-99915f4ee5e1.png)
my rabbitmq version is 3.8.22 , I'm not sure if I can solve my problem by upgrading the latest version immediately.

**Comment by lukebakken** [comment]
Please export your definitions and _attach_ them to your response.

**Reply by dormanze** [reply]
Thank you for your reply. The attachment is my definitions.
The name of the queue in question is : **mateinfo.system.plane-channel-msg-adc-management.adc-mgt-system-adc-mgt-system-6855cf9c88-85j2n-172.18.0.205-1624113247**
The name of the routing in question is :  **system.plane-channel-msg-adc-management**
[rabbitmqservice.zip](https://github.com/rabbitmq/rabbitmq-server/files/10088057/rabbitmqservice.zip)

**Reply by lukebakken** [reply]
There is only one binding for that queue in the definitions:

```
    {
      "source": "mateinfo.temp.topic",
      "vhost": "/",
      "destination": "test_queue_02",
      "destination_type": "queue",
      "routing_key": "system.plane-channel-msg-adc-management",
      "arguments": {}
    }
```

In order for us to assist you, we need an _exact_ sequence of commands to run to observe the issue you report. Ideally, this would be in a script that runs [`rabbitmqadmin`](https://raw.githubusercontent.com/rabbitmq/rabbitmq-server/main/deps/rabbitmq_management/bin/rabbitmqadmin) to declare exchanges, queues and bindings in such a way to demonstrate this issue.

It is *impossible* for us to assist you without being able to reproduce this issue. We do not have time to guess how to do it.

After reading #4237 I don't think that is relevant as you do not seem to be using exclusive or auto-delete queues, nor are declaring / deleting them rapidly.

**Reply by dormanze** [reply]
Thank you for your support and quick reply.
My problematic queue is also exclusive and auto-delete queues.You can see the binding relationship from the details of the queue.
There is not found binding for ``` mateinfo.system.plane-channel-msg-adc-management.adc-mgt-system-adc-mgt-system-59f45777fd-vt6kp-172.18.0.205--644182551 ```  in the definitions but this binding relationship can be found on the UI and can be found on ``` rabbitmqctl list_bindings ``` .
![image](https://user-images.githubusercontent.com/37854724/204070536-902d9875-a18b-4cde-be53-03637df7dc3d.png)
![image](https://user-images.githubusercontent.com/37854724/204069867-0b324175-aee2-415d-94be-80f60f9d75be.png)

**Reply by lukebakken** [reply]
Please carefully re-read my reply. Without a way to reproduce the issue, we can't assist. We don't have the time to guess how to reproduce this issue.

If you can provide _exact_ steps to reproduce this state, please do so.

**Comment by michaelklishin** [comment]
RabbitMQ 3.8 has been [out of support](https://www.rabbitmq.com/versions.html) since July 31st.

**Reply by dormanze** [reply]
Thank you for your reply.
I refer to the discussions #4237 and find that this problem also exists in 3.9.10.
So I'm not sure if upgrading to the latest version can solve my problem.
